p2p/src/peer_manager/mod.rs

```diff
@@ -65,8 +60,9 @@ use self::{
     peerdb::storage::PeerDbStorage,
 };
 
-/// Maximum number of connections the [`PeerManager`] is allowed to have open
-const MAX_ACTIVE_CONNECTIONS: usize = 128;
+/// Maximum number of outbound connections the [`PeerManager`] is allowed to have open
+/// This value is constant because users should not change this.
+const MAX_OUTBOUND_CONNECTIONS: usize = 8;
```

Comment:
Two things:

1. I think bitcoin does 16.
2. I hope this doesn't prevent users from choosing to connect to someone if the max is hit. A user should be able to choose to connect to peers through RPC.

Reply:
> I think bitcoin does 16. Right?

No, bitcoin does 8. From https://github.com/bitcoin/bitcoin/blob/519ec2650e7a4dd2850ff74cba403cf446a44a5a/src/net.h#L68:

```
/** Maximum number of automatic outgoing nodes over which we'll relay everything (blocks, tx, addrs, etc) */
static const int MAX_OUTBOUND_FULL_RELAY_CONNECTIONS = 8;
```

> I hope this doesn't prevent users from choosing to connect to someone if the max is hit. A user should be able to choose to connect to peers through RPC.

`MAX_OUTBOUND_CONNECTIONS` is not checked from RPC connect command, so users should be able always connect with RPC.
BTW, should RPC connect work the same way like the `addnode` flag? If yes, then we would need to add/update the address in `PeerDb` with the `user_added` flag set. Right now the node will try to connect just once after the RPC request. I guess we could check how bitcoin works.
And I just noticed that the node won't enforce connections to added addresses (I copied the code from the DNS server where it's not a problem because there is no limit on outbound connections). I will add TODO for this issue.

Reply:
Connecting from RPC will need a little more work, because we want to make it possible to maintain connections with a specific peer. So it's a superset of what `addnode` does in bitcoin. Substrate/Parity has 3 types of peers for this.

There's an issue about this.

Reply:
OK, sure, I will add this. But let's please do this in the next PR.

Reply:
Sure.